src/anchoredinterval.jl

```diff
@@ -58,8 +58,41 @@ AnchoredInterval{5 minutes,DateTime,Closed,Closed}(2016-08-11T12:30:00)
 
 See also: [`Interval`](@ref), [`HE`](@ref), [`HB`](@ref)
 """
-struct AnchoredInterval{P, T, L <: Bound, R <: Bound} <: AbstractInterval{T,L,R}
+struct AnchoredInterval{P, T, L <: Bounded, R <: Bounded} <: AbstractInterval{T,L,R}
     anchor::T
+
+    function AnchoredInterval{P,T,L,R}(anchor::T) where {P, T, L <: Bounded, R <: Bounded}
+        # A valid interval requires that neither endpoints or the span are nan. Typically,
+        # we use `left <= right` to ensure a valid interval but for `AnchoredInterval`s
+        # computing the other endpoint requires `anchor + P` which may fail with certain
+        # types (e.g. ambiguous or non-existent ZonedDateTimes).
+        #
+        # We can skip computing the other endpoint if both the anchor and span are finite as
+        # this ensures the computed endpoint is also finite.
+        if !_isfinite(anchor) || !_isfinite(P)
+            left, right = sign(P) < 0 ? (anchor + P, anchor) : (anchor, anchor + P)
+
+            if !(left <= right)
+                msg = if sign(P) < 0
+                    "Unable to represent a right-anchored interval where the " *
+                    "left ($anchor + $P) > right ($anchor)"
+                else
+                    "Unable to represent a left-anchored interval where the " *
+                    "left ($anchor) > right ($anchor + $P)"
+                end
+                throw(ArgumentError(msg))
+            end
+        end
+
+        return new{P,T,L,R}(anchor)
+    end
+end
+
+_isfinite(x) = iszero(x - x)
+_isfinite(x::Real) = Base.isfinite(x)
```

Comment:
I'm attempting to get this functionality added to base: https://github.com/JuliaLang/julia/pull/36380